File: sdk-php/src/UsageCallbackResult.php
<?php

declare(strict_types=1);

namespace Tollara\AgentSdk;

final class UsageCallbackResult
{
    public function __construct(
        public bool $success,
        public int $httpStatus,
        public string $httpStatusText,
        public string $requestUrl,
        public ?string $responseBody = null,
        public ?string $networkError = null,
    ) {
    }
}

File: sdk-java/src/main/java/com/tollara/client/model/UsageCallbackResult.java
package com.tollara.client.model;

import lombok.Builder;
import lombok.Value;

/**
 * Result of a progress or completion callback POST to the usage service.
 */
@Value
@Builder
public class UsageCallbackResult {
    boolean success;
    int httpStatus;
    String httpStatusText;
    String requestUrl;
    String responseBody;
    String networkError;
}
